content/articles/2019-08-lifecycle-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,255 @@
+  warning = colourise_chunk,
+  error = colourise_chunk
+)
+
+options(crayon.enabled = TRUE)
+```
+
+
+It is with unmeasured exhilaration that we announce the release of [lifecycle 0.1.0](https://lifecycle.r-lib.org) on CRAN. lifecycle is a toolkit for managing the lifecycle of your exported functions with shared conventions, documentation badges, and non-invasive deprecation warnings.
+
+The main goal of lifecycle is to streamline how the evolution of your exported API is communicated to your users. It achieves this by:
+
+* Defining a set of stages that a function or argument can be in.
+
+* Providing badges for each lifecycle stage that you can insert in your documentation.
+
+* Providing a set of functions to signal deprecation warnings with increasing levels of non-invasive verbosity.
+
+We started using these tools and conventions in r-lib and tidyverse packages a few months ago. We now make them available for all package developers. Install the package from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("lifecycle")
+```
+
+In this post, we'll briefly present the lifecycle workflow. Read the [Get started](http://lifecycle.r-lib.org/articles/lifecycle.html) vignette for a more complete description. Normally you'd call lifecycle functions with the namespace prefix, but for the purposes of illustrating the package we're attaching the package with `library()`:
+
+```{r, message = FALSE}
+library(lifecycle)
+```
+
+
+## Lifecycle stages
+
+The stages for functions and arguments are modelled after the [lifecycle stages for packages](https://www.tidyverse.org/lifecycle/).  There are 4 __development__ stages. Then, if the decision of discontinuing a feature has been made, it goes through 3 __deprecation__ stages.
+
+### Development stages
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-experimental.svg")
+```
+
+An __experimental__ feature is in the very early stage of development. It is exported so users can start to use it and report feedback, but its interface and/or behaviour is likely to change in the future. It is generally best to avoid depending on experimental features.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-maturing.svg")
+```
+
+The interface and behaviour of a __maturing__ feature has been roughed out, but finer details are likely to change. It still needs more feedback to find the optimal API.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-stable.svg")
+```
+
+A feature is considered __stable__ when the author is happy with its interface and behaviour. Major changes are unlikely, and breaking changes will occur gradually, through a deprecation process.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-questioning.svg")
+```
+
+In the __questioning__ stage, the author is no longer convinced that the feature is the optimal approach. However, there are no recommended alternatives yet.
+
```

Comment:
It might be worth a paragraph that tries to explain a bit more of the reasoning behind these stages: i.e. we're trying to get the right balance between exposing the function too few people (so we don't get enough data about what works and what's missing) and exposing it to too many people (so people start using something that isn't fully baked).